### PR TITLE
Test async communication

### PR DESCRIPTION
**Pull request by wzhongyuan**
## What changes were proposed in this pull request?

(Please fill in changes proposed in this patch)

## How was this patch tested?

(Please explain how this patch was tested. E.g. unit tests, integration tests, manual tests)
(If it is possible, please attach a screenshot; otherwise, remove this)

## Related links or issues (optional)
fixed https://github.com/intel-analytics/BigDL/issues/XXX

### CODE REVIEW

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/Test.scala [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+
+    logger.info(s"total nodes : ${nodes}")
+
+    logger.info(s"cores per node : ${cores}")
+
+    val totalLen = 26214400
+
+    val splits = args(0).toInt
+
+    val comms = sc.parallelize((0 until nodes), nodes).mapPartitions(p => {
```
Comment (yiheng) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/Test.scala]: are you sure they're on different nodes?

**Comment by jason-dai** [comment]
Please add detailed description for the PR

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/Test.scala [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+
+class AsyncComm(splits: Int, partition: Int, partitionId: Int) extends Serializable {
+  val executorService = Executors.newFixedThreadPool(10)
+  def sync(input: Tensor[Float]): Unit = {
+    val tsize = input.nElement()
+    val step = tsize / splits
+    var futures = new Array[Future[Tensor[Float]]](splits)
+    var i: Int = 0
+    while (i < splits) {
+      val viewTensor = Tensor(input.storage(), tsize - step + 1)
```
Comment (yiheng) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/Test.scala]: looks like this API will create a tensor with storage.size

Comment (yiheng) [reply spark/dl/src/main/scala/com/intel/analytics/bigdl/Test.scala]: Tensor(step)?

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/Test.scala [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+      i += 1
+    }
+    futures.foreach(f => f.get())
+  }
+
+  class SyncTask(input: Tensor[Float], partition: Int, split: Int) extends Callable[Tensor[Float]] {
+    override def call(): Tensor[Float] = {
+      val curr = System.currentTimeMillis()
+      BlockManagerWrapper.putSingle(getBlockId("SyncTask" + split + partitionId),
+        input, StorageLevel.MEMORY_ONLY_SER)
```
Comment (yiheng) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/Test.scala]: MEMORY_ONLY_SER?

**Comment by wzhongyuan** [comment]
@jason-dai   this PR will not be merged, it's for testing purpose and used for offline review :)

**Pull request closed without merging**